cron-ci/build-hpc-stack.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash -l
+# pass -l (treat script as login shell) so lmod module environment is picked up
+set -eu
+
+# mm-dd-yyy-hh:mm
+hpc_logdate=$(date +'%m-%d-%Y-%R')
+hpc_logname=hpc-stack_${hpc_logdate}.log
+hpc_log=${LOG_PATH}/${hpc_logname}
+
+# Parse hpc-stack yaml file to get version variables
+source stack_helpers.sh
+eval $(parse_yaml ${HPC_STACK_FILE} "STACK_")
+
+# get the current hpc-stack hash and compare it to the previously built hash (if it exists)
+# if it exists and matches current hash, don't build
+current_hash=$(git rev-parse HEAD)
+
+build=true
+if [[ -f "${HPC_HOMEDIR}/prev_hash.txt" ]]; then
+    prev_hash=$(cat ${HPC_HOMEDIR}/prev_hash.txt)
+    if [[ "$current_hash" == "$prev_hash" ]]; then
+        build=false
+        echo `date`
+        echo ""
+        echo "hpc-stack version has not changed since last time. Not building."
+        echo "hpc-stack hash: ${current_hash}"
+    fi
+fi
+
+if [[ "$build" == true ]]; then
+
+    # Don't build these libs because NOAA RDHPCS systems can't use wget or connect to gitlab
+    # finds the name of the library and changes the next line to "build: NO"
+    sed -i '/^udunits/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^szip/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^nccmp/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^cdo/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^boost/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+    sed -i '/^eigen/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
+
+    # pass no to setup_modules when asked for native compiler/mpi. Using modules.
+    yes no | ./setup_modules.sh -c ${HPC_CONFIG} -p ${HPC_INSTALL_PATH} >> ${hpc_log} 2>&1
+
+    ./build_stack.sh -c ${HPC_CONFIG} -p ${HPC_INSTALL_PATH} -y ${HPC_STACK_FILE} -m >> ${hpc_log} 2>&1
+
+    # Edit stack file to build debug version of ESMF
+    # Turn everything to build: NO, then enable just ESMF, then set debug: YES
+    sed -i 's/build: YES/build: NO/' ${HPC_STACK_FILE}
+    sed -i '/^esmf/{n;s/.*/  build: YES/}' ${HPC_STACK_FILE}
+    sed -i 's/debug: NO/debug: YES/' ${HPC_STACK_FILE}
+
+    # Build debug version of ESMF
+    ./build_stack.sh -c ${HPC_CONFIG} -p ${HPC_INSTALL_PATH} -y ${HPC_STACK_FILE} -m >> ${hpc_log} 2>&1
+
+    # save the current hash as the previous hash to compare to next time
+    echo $current_hash > $HPC_HOMEDIR/prev_hash.txt
+
+    echo `date`
+    echo ""
+    echo "hpc-stack hash: ${current_hash}"
+    echo "hpc-stack log: ${hpc_log}"
+    echo ""
+
+     # check if hpc-stack build succeded 
+    if grep -qi "build_stack.sh: SUCCESS!" ${hpc_log}; then
+        echo "hpc-stack build: PASS"
+    else
+        echo "hpc-tack build: FAIL"
+        exit 1
+    fi
+fi
```

Comment:
While this is perfectly acceptable, I prefer not using big if-else-blocks.  Here is an alternative:
```
[[ "$build" == false ]] && ( echo "Nothing changed, so nothing to build"; exit 0 )

# Don't build these libs because NOAA RDHPCS systems can't use wget or connect to gitlab
# finds the name of the library and changes the next line to "build: NO"
sed -i '/^udunits/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
sed -i '/^szip/{n;s/.*/  build: NO/}' ${HPC_STACK_FILE}
...
```